### ISSUE TITLE
[FileDrop] Deprecate in favor of FileUploader from Carbon

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon-addons-iot-react#611]
Carbon [recently added drop and drop file uploader components (`#3872`)](https://github.com/carbon-design-system/carbon/pull/3872). It seems this may invalidate the need for us to have our own version of this.

We should evaluate and consider deprecating [our FileDrop](https://carbon-addons-iot-react.com/?path=/story/watson-iot-filedrop--browse) in favor of the Carbon component, or extending the carbon component with any additional functionality we require.

@lukefirth @scottdickerson Do you see any reason we couldn't deprecate ours?

**Comment by tay1orjones** [comment]
Worth noting the new components are available in Carbon v10.7. We could deprecate at the same time as the upgrade in #581

**Comment by tay1orjones** [comment]
@bryancboyd could you add some context here as to your thoughts on what we can/can't deprecate?

**Comment by davidicus** [comment]
this issue is tale. Closing but feel free to re-open if necessary

**Comment by tay1orjones** [comment]
I think we should reopen the conversation about deprecating FileDrop, with the aim to consolidate on the stock Carbon component, FileUploader

**Comment by sstone2423** [comment]
I don't think this would affect Monitor too much. We only use FileDrop once it could easily be converted to use FileUploaderButton instead. Any thoughts @scottdickerson ?

**Comment by lukefirth** [comment]
Agreed - in fact, I strongly suspect we ended up with our own due to the fact that Carbon made this available in the sketch kit a while before they made it available as a component in code, rather than us having any real desire to have our own.

(In fact - from a design kit side, we don't actually have our own version of this, we just use the Carbon one).

My vote: Deprecate away 😄

**Comment by dianatran18** [comment]
I'm good with deprecating if it doesn't affect any teams using it. From all of the design use cases I've heard of, the Carbon FileUploader was sufficient to use

**Comment by stuckless** [comment]
FYI... Puralotor just stopped by and picked up the ceiling tiles.
